aspnet/mvc/models/validation.rst

```diff
@@ -1,8 +1,211 @@
+
+.. literalinclude:: validation/sample/ClassicMovieAttribute.cs
+ :language: c#
+ :lines: 39-44
+ :dedent: 8
+ 
+Attributes that implement this interface can add HTML attributes to generated fields. Examining the output for the ``ReleaseDate`` element reveals HTML that is similar to the previous example, except now there is a ``data-val-classicmovie`` attribute that was defined in the ``GetClientValidationRules`` method of ``IClientModelValidator``.
+
+.. code-block:: html
+   
+  <input class="form-control" type="datetime" 
+  data-val="true" 
+  data-val-classicmovie="Classic movies must have a release year earlier than 1960" 
+  data-val-required="The ReleaseDate field is required." 
+  id="ReleaseDate" name="ReleaseDate" value="" />
+
+Unobtrusive validation uses the data in the ``data-`` attributes to display error messages. However, jQuery doesn't know about rules or messages until you add them to jQuery's ``validator`` object. This is shown in the example below that adds a method named ``classicmovie`` containing custom client validation code to the jQuery ``validator`` object. 
+
+.. code-block:: javascript
+   
+  $(function () {
+      jQuery.validator.addMethod('classicmovie',
+      function (value, element, params) {
+	      // custom validation code
+          return false;
+      }, '');
+
+      jQuery.validator.unobtrusive.adapters.add('classicmovie',
+      function (options) {
+          options.rules['classicmovie'] = {};
+          options.messages['classicmovie'] = options.message;
+      });
+  }(jQuery));
+
+Now jQuery has the information to execute the custom JavaScript validation as well as the error message to display if that validation code returns false. 
+
+Remote validation
+-----------------
+
+Remote validation is a great feature to use when you need to validate data on the client against data on the server. For example, your app may need to verify whether an email or user name is already in use, and it must query a large amount of data to do so. Downloading large sets of data for validating one or a few fields consumes too many resources. It may also expose sensitive information. An alternative is to make a round-trip request to validate a field.
+
+You can implement remote validation in a two step process. First, you must annotate your model with the ``[Remote]`` attribute. The ``[Remote]`` attribute accepts multiple overloads you can use to direct client side JavaScript to the appropriate code to call. The example points to the ``VerifyEmail`` action method of the ``Users`` controller. 
+
+.. literalinclude:: validation/sample/User.cs
+ :language: c#
+ :lines: 6-24
+ 
+The second step is putting the validation code in the corresponding action method as defined in the ``[Remote]`` attribute. It returns a ``JsonResult`` that the client side can use to proceed or pause and display an error if needed.
+ 
+.. literalinclude:: validation/sample/UsersController.cs
+ :language: c#
+ :lines: 3-9
```

Comment:
You need a larger snippet of the `UsersController` class, not just the first few lines.